modules/multiplayer/scene_rpc_interface.cpp

```diff
@@ -73,14 +73,24 @@ int get_packet_len(uint32_t p_node_target, int p_packet_len) {
 	}
 }
 
+bool SceneRPCInterface::_sort_rpc_names(const Variant &p_l, const Variant &p_r) {
```

Comment:
You could probably use `StringName::AlphCompare` for this instead of implementing something similar ad hoc?
https://github.com/godotengine/godot/blob/83d54ab2ad476ae265b323c2b88f4623b922f4c6/core/string/string_name.h#L167

Reply:
Yeah, I looked into that, the problem is that AlphCompare expects `const StringName &` because it's meant for internal (templated) data structures (`List`/`HashMap`/etc) which implements `sort_custom` as a template function while `Array.sort_custom` expects a callable because it's a bound data type (and the callable takes `const Variant &` arguments).

So I don't think there's a way to reuse it directly. Maybe, we could use it inside `_sort_rpc_names`  when we detect StringName and get (maybe?) a performance optimization? (I don't think the compiler is smart enough to optimize away the string creation otherwise)

Reply:
Makes sense. BTW I wondered why we're using Variant for the rpc config and the names instead of explicitly Dictionary/String(Name)?

Reply:
That's a good question, IIRC originally that was to allow migrating from a dictionary to a refcounted/struct and handle compatibility internally without changing the function signature (that was before we had compatibility methods and so on).

I think at this point we could switch to dictionary (but then, the config of each method would still be composed of "variants" since that's what dictionary values are).

We could even switch to something like `HashMap<StringName,InternalStruct>` but that involves doing copies every time we cross the boundaries (e.g. gdextension ScriptExtension._get_rpc_config), and I'd rather do that in the multiplayer module itself where they can be properly cached.